src/identity.js

```diff
@@ -0,0 +1,56 @@
+
+const isFunction = require('./utils/is-function')
+
+class Identity {
+  constructor (id, publicKey, provider) {
+    if (!id) {
+      throw new Error('Identity id is required')
+    }
+
+    if (!publicKey) {
+      throw new Error('Invalid public key')
+    }
+
+    if (!provider) {
+      throw new Error('Identity provider is required')
+    }
+
+    if (!provider.sign) {
```

Comment:
I feel it's not the responsibility of the *Identity* to check the providers shape/functions/functionality. Not doing that keep the responsibilities clear and improves decoupling the Identity with the IdentityProvider.